### PR TITLE
Add reduced diag precision input parameter

### PR DESCRIPTION
**Pull request by dpgrote**
This adds a parameter that allows the user to control the precision of the data written to the reduced diagnostics output files.

### CODE REVIEW

File: Source/Diagnostics/ReducedDiags/ReducedDiags.H [hunk 1]
```diff
@@ -42,6 +42,9 @@ public:
     /// separator in the output file
     std::string m_sep = " ";
 
+    /// precision for data in the output file
+    int m_precision = 14;
```
Comment (lucafedeli88) [inline Source/Diagnostics/ReducedDiags/ReducedDiags.H:46]:
```suggestion
    constexpr int default_precision = 14;
    int m_precision = default_precision;
```
I suggest this small change to improve readability.

**Review by lucafedeli88 — COMMENTED** [review]
Thanks for this PR! I left a tiny comment

File: Source/Diagnostics/ReducedDiags/ReducedDiags.H (hunk 1 above)
Comment (dpgrote) [reply Source/Diagnostics/ReducedDiags/ReducedDiags.H:46]: I can make this change, but note that none of the other variables have a separate constexpr to hold the default value.

Comment (lucafedeli88) [reply Source/Diagnostics/ReducedDiags/ReducedDiags.H:46]: You are right. We should maybe address these issues all at once if at some point we decide to enforce the `readability-magic-numbers` check in `clang-tidy` !

**Review by lucafedeli88 — APPROVED** [review]

**Pull request merged**